File: src/simulation/common/PolarCoordinate.ts
let count = 0;

export class PolarCoordinate {
  private id: number = 0;
  private name: string = '';
  private theta: number;
  private r: number;
  private attraction: boolean = true;

  constructor();
  constructor(r: number, theta: number);
  constructor(r?: number, theta?: number) {
    if (r !== undefined && theta !== undefined) {
      count++;
      this.id = count;
      this.r = r;
      this.theta = theta;
    } else {
      this.r = 0;
      this.theta = 0;
    }
  }

  getR(): number {
    return this.r;
  }

  getTheta(): number {
    return this.theta;
  }

  setR(r: number): void {
    this.r = r;
  }

  setTheta(theta: number): void {
    this.theta = theta;
  }

  setName(name: string): void {
    this.name = name;
  }

  getName(): string {
    return this.name;
  }

  setAttractionFalse(): void {
    this.attraction = false;
  }

  getAttraction(): boolean {
    return this.attraction;
  }

  getId(): number {
    return this.id;
  }

  setId(id: number): void {
    this.id = id;
  }

  toString(): string {
    return 'r : ' + this.r + ' ' + 'theta: ' + this.theta;
  }
}

File: java/simulation/common/PolarCoordinate.java
package simulation.common;

public class PolarCoordinate {

private static int count = 0;
//@SuppressWarnings("unused")
private int id = 0;
//@SuppressWarnings("unused")
private String name = "";
private int theta;
private float r;
private boolean attraction = true; 


public PolarCoordinate(){}

public PolarCoordinate(float r, int theta){
	count++;
	id = count;
	this.r = r;
	this.theta = theta;
}

public float getR() {
	return r;
}

public int getTheta() {
	return theta;
}

public void setR(float r){
	this.r = r;
}

public void setTheta(int theta){
	this.theta = theta;
}

public void setName(String name){
	this.name = name;
}

public String getName(){
	return name;
}

public void setAttractionFalse(){
	attraction = false;
}

public boolean getAttraction(){
	return attraction;
}
public int getId() {
	return id;
}

public void setId(int id) {
	this.id = id;
}

@Override
public String toString() {
  return "r : " + r + " " + "theta: " + theta;
}
}